server/src/main/java/org/opensearch/extensions/ExtensionsManager.java

```diff
@@ -502,21 +502,23 @@ public void beforeIndexRemoved(
                                     new IndicesModuleRequest(indexModule),
                                     acknowledgedResponseHandler
                                 );
-                                inProgressIndexNameFuture.whenComplete((r, e) -> {
-                                    if (e != null) {
-                                        inProgressFuture.complete(response);
-                                    } else if (e == null) {
-                                        inProgressFuture.completeExceptionally(e);
-                                    }
-                                });
-                            } catch (Exception ex) {
-                                inProgressFuture.completeExceptionally(ex);
+                                inProgressIndexNameFuture.orTimeout(EXTENSION_REQUEST_WAIT_TIMEOUT, TimeUnit.SECONDS).join();
+                            } catch (CompletionException e) {
+                                if (e.getCause() instanceof TimeoutException) {
+                                    logger.info("No response from extension to request.");
+                                }
+                                if (e.getCause() instanceof RuntimeException) {
+                                    throw (RuntimeException) e.getCause();
+                                } else if (e.getCause() instanceof Error) {
+                                    throw (Error) e.getCause();
```

Comment:
We can avoid this throw and let the next else do the job

Reply:
Are you sure? This sequence was suggested by reta here: https://github.com/opensearch-project/OpenSearch/pull/5646#discussion_r1070197601